src/routes/resource.js

```diff
@@ -107,5 +114,57 @@ module.exports = function resource (endpoint, exclude, associations) {
     });
   });
 
+  // Transaction method that links all the associated models in the body
+  // for an specific model. It returns a promise that resolves with the linked
+  // instance
+  const linkAssociations = (models, req, transaction, instance) => {
+    return new Promise((resolve, reject) => {
+      if (associations.length <= 0) {
+        // Nothing to link
+        return resolve(instance);
+      }
+
+      let promises = [];
+      associations.forEach(name => {
+        if (!req.body[name]) {
+          return;
+        }
+        const body = req.body[name];
+        const link = linkInstance(instance, models, name, body, transaction);
+        promises.push(link);
+      });
+
+      Promise.all(promises).then(() => {
+        resolve(instance);
+      }).catch(err => {
+        reject(err);
+      });
+    });
+  };
+
+  // Finds or create an specific entity, and associate it to the given instance.
+  const linkInstance = (instance, models, name, entity, transaction) => {
+    const model = models[name];
+    const id = entity[CONST.iotId] || -1;
+    Reflect.deleteProperty(entity, 'id');
+    return new Promise((resolve, reject) => {
+      model.findOrCreate({
+        where: { id: id },
+        defaults: entity,
+        transaction: transaction
+      }).spread(foundOrCreated => {
+        instance['set' + name](foundOrCreated, {
```

Comment:
Note that this won't always be `setAssociation`. It depends on the body content and on the type of association. For example, it could be `addAssociation` for a HasMany association.